pkg/apis/core/validation/validation.go

```diff
@@ -6329,3 +6337,20 @@ func isHeadlessService(service *core.Service) bool {
 		len(service.Spec.ClusterIPs) == 1 &&
 		service.Spec.ClusterIPs[0] == core.ClusterIPNone
 }
+
+// validateUpgradeDowngradeLoadBalancerClassName validation for loadBalancerClassName in cases of user upgrading or downgrading service
+func validateUpgradeDowngradeLoadBalancerClassName(oldService, service *core.Service) field.ErrorList {
+	allErrs := make(field.ErrorList, 0)
+	// bail out early for non LoadBalancer type of service
+	if oldService.Spec.Type != core.ServiceTypeLoadBalancer || service.Spec.Type != core.ServiceTypeLoadBalancer {
```

Comment:
This isn't right?  If I change from ClusterIP to NodePort, you need to validate that this new field is _not_ set.

```
if new.type = LB && old.type == LB {
  // ensure class did not change (if that is what we want, see other comments)
} else if new.type == LB {
  if new.class != nil {
    validate new.class
  }
} else {
  if new.class != nil {
    // error
  }
}
```

You can use the same function for create and update by passing nil for "old" and just handling that.